app/controllers/dashboard_controller.rb

```diff
@@ -4,5 +4,16 @@ class DashboardController < ApplicationController
   skip_after_action :verify_pundit_authorization
 
   def index
+    @contracts_pending = Contract.where(id: policy_scope(Contract).select(:id))
+                                 .where(status: :pending)
+                                 .group(:status)
+                                 .count
+
+    @contracts_expired = Contract.where(id: policy_scope(Contract).select(:id))
+                                 .where(state: [ :active, :expired ])
+                                 .group(:state)
+                                 .count
+
+    @contracts_data = @contracts_pending.merge(@contracts_expired) { |key, pending_count, expired_count| pending_count + expired_count }
```

Comment:
Instead of `Contract.where(id: policy_scope(Contract).select(:id))` do just `policy_scope(Contract)` as this is what policy_scope is doing behind the scenes.

Reply:
Also maybe the merge should be done in the activerecord query instead of as two seperate queries.